src/main/java/digital/slovensko/autogram/ui/gui/MainMenuController.java

```diff
@@ -110,10 +110,15 @@ private void signFiles(List<File> list) {
         var filesList = getFilesList(list);
         if (filesList.size() == 1) {
             var file = filesList.get(0);
-            var job = SigningJob.buildFromFile(file,
-                    new SaveFileResponder(file, autogram, userSettings.shouldSignPDFAsPades()),
-                    userSettings.isPdfaCompliance(), userSettings.getSignatureLevel(), userSettings.isEn319132(), tspSource, userSettings.isPlainXmlEnabled());
-            autogram.sign(job);
+            var document = SigningJob.createDSSFileDocumentFromFile(file);
```

Comment:
Nebolo by lepšie to volať v tejto metóde?

Reply:
a prečo to nedáme do SigningJob.build? To sa volá aj pre buildFromRequest aj pre buildFromFile.

Reply:
keď potrebuješ tie getParametersForFile, tak by som to normláne osobitne zavolal v buildFromFile a buildFromRequest - nevadí, že to bude 2x, stále to bude lepšie ako teraz volané z GUI.

A neviem úplne, čo sa stane, keď pošleš taký súbor cez API - nebude už server chcieť niečo čítať z toho súboru?

Reply:
Viem, že som to včera nespravil tak lebo `SigningJob#build` vracia `SigningJob` a ja `Autogram#handleProtectedPdfDocument` musím obaliť do work threadu, teda tak či tak by sme to museli ešte vyššie obaliť v tomto jednom prípade ked signujeme jeden súbor, keďže pri hromadnom to obalené už je a pri CLI to netreba. Avšak bolo by to asi lepšie, zmazal by sa duplikačný kód 👍

Reply:
> keď potrebuješ tie getParametersForFile, tak by som to normláne osobitne zavolal v buildFromFile a buildFromRequest - nevadí, že to bude 2x, stále to bude lepšie ako teraz volané z GUI.
> 
> A neviem úplne, čo sa stane, keď pošleš taký súbor cez API - nebude už server chcieť niečo čítať z toho súboru?

Ešte stále je to draft, API budem riešiť dnes 😅

Reply:
Nemáš nejaký lepší nápad ako sa zbaviť tohto?
https://github.com/slovensko-digital/autogram/pull/498/files#diff-daf36d389667b41baf5336c039d7fc72f9132298c577ef4ec11e45dd5fe54071R114

Reply:
moc pekné riešenie nemám. Ale teda dá sa poslať GUI inštancia do MainMenuControllera a potom volať rovno gui.onWorkThreadDo a nemusíš tam mať tú obaľovaciu metódu cez Autogram.

Reply:
ale teda tento `handleProtectedPdfDocument` bude musieť ostať takto ešte pred vzniknutím SigningJobu, lebo obsah pdf potrebujeme čítať ešte pri vyrábaní parametrov. Otázka je, či to vyhodíme z tohto buildu a spravíme to ešte niekde predtým.

Reply:
@xhyrom toto si nepozeral? Presunutie toho "buildFromFile" do Autogramu.